uaclient/config.py

```diff
@@ -236,15 +240,43 @@ def _status(self) -> "Dict[str, Any]":
         return response
 
     def status(self) -> "Dict[str, Any]":
-        """Return status as a dict, using a cache for non-root users"""
-        if os.getuid() == 0:
-            status = self._status()
-            self.write_cache("status-cache", status)
-            return status
-        cached_status = cast("Dict[str, Any]", self.read_cache("status-cache"))
-        if not cached_status:
-            return DEFAULT_STATUS
-        return cached_status
+        """Return status as a dict, using a cache for non-root users
+
+        When unattached, get available resources from the contract service
+        to report detailed availability of different resources for this
+        machine.
+
+        Write the status-cache when called by root.
+        """
+        from uaclient.contract import get_available_resources
+        from uaclient.entitlements import ENTITLEMENT_CLASS_BY_NAME
+
+        if not self.is_attached:
```

Comment:
Dan, is the approach of handling unattached case up in status instead of down in _status distasteful?  I only pulled it up because non-root users running status have to do the same work a root  user does